add CountryCode property

othmane-fadili opened the pull request.

jlhughes commented on deployment/https/services.jrl:
```diff
@@ -1,4 +1,4 @@
-p({
+\p({
```
```suggestion
p({
```

jlhughes commented:
What is this branch for?  It is tracking all changes and generates lots of notifications.

kgrgreer commented:
@othmane-fadili It likes something is wrong with the PR because it has 275 commits and 173 updated files.

kgrgreer committed:
Add TODO.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

abdelaziz-mahdy committed:
feat: add Msg parser decorator for attaching message configurations

abdelaziz-mahdy committed:
feat: enhance parser messages with flexible message attachment

kgrgreer committed:
Changes to FoamTagLoader to work better with U3.

Commit:
ReflowToolBar access using toolbarControlDAO

Nauna committed:
Merge pull request #4689 from kgrgreer/reflowToolbar_2

ReflowToolBar access using toolbarControlDAO

kgrgreer committed:
Limit max number of concurrent JS HTTPBox connections to 30 to prevent overloading server.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

abdelaziz-mahdy committed:
feat: add toProperties functions to Max and Min sinks; improve error handling in LabeledSink

kgrgreer committed:
Revert "Limit max number of concurrent JS HTTPBox connections to 30 to prevent overloading server."

This reverts commit be880de300086861b8c09e511839de5f83f6d81e.

abdelaziz-mahdy committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into add-missing-toProperties-functions

kgrgreer committed:
Fix reflow/lib.fix() method.

abdelaziz-mahdy committed:
fix(reflow): prevent double execution on load by awaiting initial execution (#4692)

abdelaziz-mahdy committed:
feat(DAOPrompt): add skipInitialReset_ property to prevent double execution on load (#4693)

abdelaziz-mahdy committed:
feat(validation): implement custom validation for required fields and track empty source values

kgrgreer committed:
Spacing.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

Commit:
Added null checks to daos in Commands.js

Nauna committed:
Merge pull request #4695 from kgrgreer/daoNullChecksCommands

Added null checks to daos in Commands.js

kgrgreer committed:
Fix AbstractFObjectFormatter.maybeAppendComma() when multi-line output is enabled.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

kgrgreer committed:
Port FOAM1 Dragon to FOAM3

kgrgreer committed:
Listen to prerequisiteCapabilityJunctionDAO to invalidate CachingAuthService.

kgrgreer committed:
Add support to async normalize() methods in the Input tag.

kgrgreer committed:
Add cache to serviceProviderDAO.

kgrgreer committed:
Formatting.

kgrgreer committed:
Temporarily disable JS MDAO In and Contains indexing which are broken because MergePlan doesn't work.

kgrgreer committed:
Remove stale comment.

kgrgreer committed:
PT-FIXES: DOS-1889 Search operator is removed automatically on first selection when choosing an operator from the Search list. Adds QueryRouter to make doing full-text search possible. General TextSearchView cleanup.

VesnaSUG committed:
DOS-1996 Split JAR into two, one for binary and the other for journals (#4697)

abdelaziz-mahdy committed:
fix(date-parsing): enhance timestamp handling and validation in date parser

abdelaziz-mahdy committed:
fix: disable save button while loading flow

kgrgreer committed:
Remove dead code from CSSParser.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

kgrgreer committed:
Add support for inlined styling in Markdown HTML tags.

kgrgreer committed:
Fix pom to add QueryRouter.

kgrgreer committed:
Enable : as short-form for TEXT: in QueryRouter.

kgrgreer committed:
Fix for last QueryRouter change.

kgrgreer committed:
Enable SmartView normalization by default.

kgrgreer committed:
Spacing.

Nauna committed:
Merge pull request #4690 from kgrgreer/add-msg-decorator

PT-FIXES: DOS-1885 Add msg decorator

Nauna committed:
Merge pull request #4699 from kgrgreer/disable-save-while-loading-flow

PT-FIXES: DOS-1579 : disable save button while loading flow

jlhughes committed:
Fix https deployment: add RES_JAR_HOME to JAVA_OPTS else ResourceStorage looks in the wrong jar file

aiokaizen committed:
Fix scroll animation glitch with code editor (#4700)

Co-authored-by: Mouad Kommir <mouad.kommir@payticconnect.com>

kgrgreer committed:
Spacing.

kgrgreer committed:
Make Java Count.put() faster.

adamvy committed:
Merge adam's develoment branch (#4701)

* fix foam tag loader for u3 when the object isn't a view

* tidy up some bad formatting

* PredicatedListener in java

* Remove unused pre/post support for DAOSelectNode/FunctionNode

* Element: Observe subtrees for default mutation observer

* parsers: Fix getLastError(), remove unused ps/lastStart properties

Update flow document parser to use it properly.

* Add missing PredicatedListener file to pom

* Add java properties for enabling verbose debug logging in StandardAuthorizer and SessionServerBox

* Add support to dynamic functions for providing the slots needed, similar to this.slot(, foo$, bar$)

* Element.initKeyboardShortcuts: Don't use parentNode as the target if parent is a function node (comment node).

Otherwise this throws an error as you can't add tabIndex to an html comment

* OAuthWebAgent: Support sign up by default.

Set the username to email if not provided in state parameters

* PredicatedListener: Remove weird exception suppression

* fix typo in flow Document getLastError() call

* OAuthWebAgent tidy up username logic

abdelaziz-mahdy committed:
feat: add detachFlowChild method and improve flow child removal logic (#4474)

* feat: add detachFlowChild method and improve flow child removal logic

* Update src/foam/core/reflow/Console.js

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

* Apply suggestion from @sarthak-marwaha

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

---------

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

kgrgreer committed:
Make StandardAuthorizer more efficient by reusing Exception.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

adamvy committed:
Fix DateParser now that Grammar no longer has a ps property

adamvy committed:
More complete and cleaner fix for DateParser

aiokaizen committed:
Fix timezone issue with flow calendar sink and a problem specific to safari as well. (#4703)

Co-authored-by: Mouad Kommir <mouad.kommir@payticconnect.com>

Nauna committed:
Use full auth chain for CHECK_USER_PERMISSION (#4705)

Co-authored-by: Nauna <anna.doulatshahi@payticconnect.com>

aiokaizen committed:
Add empty value message to PieChart when no data is available (#4704)

* Add empty value message to PieChart when no data is available

* Use dynamic instead of slot

---------

Co-authored-by: Mouad Kommir <mouad.kommir@payticconnect.com>

abdelaziz-mahdy committed:
fix(tools): exit with status 1 on test failure in JavaTooling

adamvy committed:
Merge pull request #4706 from kgrgreer/exit-code-1-on-failure

jlhughes committed:
Revert "Use full auth chain for CHECK_USER_PERMISSION (#4705)"

This reverts commit cdcb22aa27af12837033be445b0f12e21ef54062.

jlhughes committed:
Merge pull request #4707 from kgrgreer/revert-4705-AuthCheckUser_update

Revert "Use full auth chain for CHECK_USER_PERMISSION"

kgrgreer committed:
Make Java mlang.Constant slightly faster.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

kgrgreer committed:
More efficient predicate merging in FilteredDAO.

abdelaziz-mahdy committed:
style(dashboard): update color variables for improved theming

abdelaziz-mahdy committed:
Fix export functionality and improve error handling

- TableColumnOutputter: Add default dateFormat factory to prevent
  undefined errors when outputter is created without dateFormat

- GoogleSheetsBasedExportDriver: Add dateFormat property and pass it
  to GoogleSheetsOutputter to fix date formatting in exports

- FLOWExportDriver: Fix flow creation
  - Change mementoStr to script (correct property name)
  - Change DAOPrompt2 to DAOPrompt (correct class name)
  - Add fallbacks for of, daoKey, plural when init() fails
  - Add label and border properties to match working flow format

- ExportModal: Add try-catch error handling to all export actions
  (convert, download, open) to display errors to users instead of
  failing silently with uncaught promise rejections

Commit:
update Permission check in UserAngGroupAuthService.checkUser

abdelaziz-mahdy committed:
Fix Import modal redirect and empty pop-up issues

- Fixed clicking Import button redirecting to main page instead of showing modal
- Fixed empty pop-up message appearing when Import modal was opened
- Fixed StackView.renderStackView() calling routeTo(undefined) when currentMemento is not set
- Fixed WizardModal to use StackBlock for proper stack navigation
- Fixed DAOBrowserView to find DAO key from delegate chain for Data Management menus
- Fixed ImportFromGoogleSheetsForm expression to handle null dao gracefully
- Fixed GoogleSheetImportModal to use dynamic check for importConfig

Root cause: StackView was calling routeTo(s.currentMemento) even when currentMemento
was undefined (disabled in Stack.js commit 6cb23bd9c9), causing redirect to main page.

jlhughes committed:
Merge pull request #4709 from kgrgreer/fix-tests

style(dashboard): update color variables for improved theming

jlhughes committed:
Merge pull request #4712 from kgrgreer/UserAndGroupAuth_update

update Permission check in UserAngGroupAuthService.checkUser

kgrgreer committed:
Java JDAO TreeIndex performance improvements.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

kgrgreer committed:
Fix Java ENUM constant creation to output correct types for arrays.

kgrgreer committed:
Add 'javaCode' property refinement to Enum values to allow per-instancew java code.

kgrgreer committed:
Temporary fix to allow for FSMs.

abdelaziz-mahdy committed:
Merge with Dev

Nauna committed:
Merge pull request #4698 from kgrgreer/fix-date-parsing-if-timestamp

PT-FIXES: DOS-2019, DOS-2008, DOS-2007 : enhance timestamp handling and validation in date parser

kgrgreer committed:
Fix java refinement for StateMachines.

Commit:
Setup a new flowMode (LIMIT_EDIT_CONSOLE & LIMIT_EDIT) transitioning between the two

Commit:
Flow Access - limit righ section based on Block settings

Commit:
add fullScreen option to flowMode.LIMIT_EDIT

Commit:
Setup limitAccess flowMode on flow block

kgrgreer committed:
Fix MarkdownView to allow html tags with numbers in them like h1 and h2.

kgrgreer committed:
Add console.error() to report why save action isn't enabled.

abdelaziz-mahdy committed:
feat: enhance NumberParser with locale-aware formatting and parsing, add NumberFormat enum, and remove obsolete tests/javacode

abdelaziz-mahdy committed:
Merge with Dev

abdelaziz-mahdy committed:
feat: add NumberFormat views and parser for locale-aware number formatting

- Introduced NumberFormatCitationView and NumberFormatSelectionView for displaying number formats with descriptions.
- Added NumberFormatRichChoiceView for handling enum values related to number formats.
- Implemented NumberGrammar and NumberParser to support parsing and formatting of standard and European number formats.
- Created comprehensive tests for NumberParser covering various formats, including edge cases and invalid inputs.
- Removed legacy NumberParser implementation and associated tests.

abdelaziz-mahdy committed:
feat: refactor FloatView to use Intl.NumberFormat for locale-aware number formatting and remove obsolete locale property

abdelaziz-mahdy committed:
feat: add lenient parsing tests for variable digit grouping and extended decimals in NumberParser

abdelaziz-mahdy committed:
fix: update NumberParserTest flags to include java&test

abdelaziz-mahdy committed:
refactor: streamline fromCSV methods in Int and Float classes for better readability

abdelaziz-mahdy committed:
fix: update copyright year to 2026 in multiple files

abdelaziz-mahdy committed:
refactor: enhance fromCSV methods in Int and Float classes with improved parameter documentation and parsing logic

Commit:
pick up flow permission to filter limitedEdit permission access

Nauna committed:
Merge pull request #4719 from kgrgreer/FlowAccess_updates

Flow access updates

adamvy committed:
Some tidying (#4718)

* fix bug in oauth user registration

* make oauth window vs redirect flow a service configuration parameter

* fix parse errors when a string pstream is created with an empty string

this.StringPStream.create({ str: "" }) results in ps.apply being undefined

* remove stale before/after for daoselectnode

kgrgreer committed:
Remove stale code.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

abdelaziz-mahdy committed:
fix: update memento handling in FilterController and FilterView, improve error logging in DeFeedback

kgrgreer committed:
Add console logging of data errors which prevent save in DAOCreateView.

abdelaziz-mahdy committed:
PT-FIXES: DOS-2071 : disable auto-scroll in presentation-only mode (#4716)

* feat: disable auto-scroll in presentation-only mode

* feat: implement selectFromTree function to enable block selection and scrolling

abdelaziz-mahdy committed:
docs: add Safari debugging guide with HTTPS instructions (#4710)

abdelaziz-mahdy committed:
fix(filter): update date range handling in DateFilterView for inclusive/exclusive filtering

abdelaziz-mahdy committed:
disable undo and redo while loading

abdelaziz-mahdy committed:
fix: ensure all pending batch operations complete before updating status in UploadSink

kgrgreer committed:
Add data error logging when save is disabled.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

Nauna committed:
update JSONFObjectFormatter outputProperty to default a class if empty object

abdelaziz-mahdy committed:
fix(formatter): conditionally output class for empty FObjects based on OutputDefaultClassNames

abdelaziz-mahdy committed:
Revert "fix(formatter): conditionally output class for empty FObjects based on OutputDefaultClassNames"

This reverts commit 9a5e0c278b2685bf5b9522775615c27dba19da5c.

abdelaziz-mahdy committed:
fix(formatter): enhance handling of empty FObjects based on OutputDefaultClassNames

abdelaziz-mahdy committed:
test(formatter): update tests for empty FObject handling with default settings

Nauna committed:
Merge pull request #4726 from kgrgreer/Write-deafult-class-fix

Write default class fix

Nauna committed:
Merge pull request #4722 from kgrgreer/fix-date-view

PT-FIXES: DOS-2100: update date range handling in DateFilterView for inclusive/exclusive filtering

Nauna committed:
Merge pull request #4724 from kgrgreer/ensure-await

fix: ensure all pending batch operations complete before updating status in UploadSink

Nauna committed:
Merge pull request #4723 from kgrgreer/disable-undo

PT-FIXES: DOS-1579 disable undo and redo while loading

Nauna committed:
add Block.ALLOW_LIMITED_EDIT through sanizitation in Console

Nauna committed:
disable 'esc' toggleMode for LIMIT_EDIT : Console

jlhughes committed:
Build Doc - add building third-party from source, and enabling java feature flags

Commit:
reset validaion in UserPropertyAvailabilityView

kgrgreer committed:
Initial check-in of LLM2.md.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

abdelaziz-mahdy committed:
fix(parse): add '+' to valid characters in QueryParser

abdelaziz-mahdy committed:
fix(parse): add '+' to valid characters in QueryParser

abdelaziz-mahdy committed:
fix(parse): add support for email plus addressing in QueryParser tests

abdelaziz-mahdy committed:
Improve tests

abdelaziz-mahdy committed:
fix(parse): update extractValues to handle In predicates for comma-separated values

Nauna committed:
Merge pull request #4731 from kgrgreer/memnto-encoding-fix

fix(parse): add '+' to valid characters in QueryParser

Nauna committed:
Merge pull request #4721 from kgrgreer/fix-filter-view-not-setting-memnto

PT-FIXES: DOS-2067 : update memento handling in FilterController and FilterView, improve error logging in DeFeedback

noodlemoodle committed:
fix filter save (#4714)

Co-authored-by: ruby <ruby.zhang@payticconnect.com>

abdelaziz-mahdy committed:
Cleanup the console.js to have enum decide what each mode does (#4730)

abdelaziz-mahdy committed:
docs: add comprehensive documentation for the SPID system (#4727)

* docs: add comprehensive documentation for the SPID system

* docs: update SPID documentation to include automatic setup with EasyDAO

* docs: refine best practices for SPID implementation and usage

* docs: clarify automatic setup process for ServiceProviderAwareDAO in EasyDAO

jlhughes committed:
JavacMaker - support building third party repositories from source. Instructions in Build.md

Nauna committed:
update efficiency on JSONFObjectFormatter.outputPropery

Nauna committed:
Merge pull request #4725 from kgrgreer/WriteDefaultClass

update JSONFObjectFormatter outputProperty to default a class if empty object

abdelaziz-mahdy committed:
feat: add immediate data parsing in init method for TextSearchView

Nauna committed:
Merge pull request #4732 from kgrgreer/fix-inital-wrong-call

PT-FIXES: DOS-2121 Fix URL search prefilter causing duplicate DAO queries

abdelaziz-mahdy committed:
fix: handle multipart keys in OverlayActionListView object fetching

Nauna committed:
Merge pull request #4733 from kgrgreer/fix-detail-view-incase-of-multipart

PT-FIXES: DOS-2126 Fix Table Row Actions for Multipart Key Models

abdelaziz-mahdy committed:
feat: enhance reflow method to support flow mode logging and dynamic routing

kgrgreer committed:
Support matching property names with literalIC() instead of literal() in SimpleQueryParser.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

kgrgreer committed:
Catch Throwable instead of IOEXception in AbstractF3FileJournal.

abdelaziz-mahdy committed:
fix: improve memento encoding and synchronization of tailStr

abdelaziz-mahdy committed:
fix(memento): prevent inline views from breaking navigation chain

   Inline views created via FObject.toE() were connecting to the root
   memento and orphaning the DAOController, causing URL updates to fail
   after the first navigation.

   - Add isInlineView_ flag to isolate inline view mementos
   - Add initializing_ flag to prevent premature URL updates on page load
   - Prevent non-router views from overwriting router tails
   - Re-enable detachTail() when route changes

abdelaziz-mahdy committed:
fix(memento): prevent inline views from breaking navigation chain

   Inline views created via FObject.toE() were connecting to the root
   memento and orphaning the DAOController, causing URL updates to fail
   after the first navigation with multipart ID records.

   - Add isInlineView_ flag to isolate inline view mementos

abdelaziz-mahdy committed:
Merge branch 'fix-back-button' of github.com:kgrgreer/foam3 into fix-back-button

abdelaziz-mahdy committed:
refactor(memento): simplify parent postSet logic and remove unused flags

abdelaziz-mahdy committed:
fix(memento): create isolated memento for inline views to prevent navigation chain issues

abdelaziz-mahdy committed:
Enable NDiff on Flow DAO and Improve Compare UI

abdelaziz-mahdy committed:
fix(ndiff): remove redundant networkTransient property

Nauna committed:
Merge pull request #4737 from kgrgreer/fix-back-button

PT-FIXES: DOS-2125 Fix memento navigation for multipart ID records

abdelaziz-mahdy committed:
refactor(ndiff): simplify NDiff methods and enhance menu configuration

abdelaziz-mahdy committed:
feat(ndiff): add projectionSafe property and enhance ComparisonView for model property comparison

abdelaziz-mahdy committed:
Merge with dev

Nauna committed:
Merge pull request #4736 from kgrgreer/fix-memnto

PT-FIXES: DOS-2080 Fix filter state loss when navigating to detail view and back

abdelaziz-mahdy committed:
fix(choice-view): prevent auto-selection of single choice when placeholder is set

kgrgreer committed:
Remove dead code.

abdelaziz-mahdy committed:
feat(reflow): add underscore-to-camelCase normalization in ColumnParser

   Support arbitrary underscore-separated column names (e.g., Element_attribute)
   by normalizing them to camelCase property names (elementAttribute). This allows
   CSV files with various naming conventions to be mapped to FOAM properties.

   - Add normalizeToPropertyName() method for underscore-to-camelCase conversion
   - Add fifth fallback matcher using action() for dynamic lookup after normalization
   - Change 'end' rule to peek() to avoid consuming delimiters
   - Add comprehensive test suite covering all matching strategies

abdelaziz-mahdy committed:
docs: add guide for using multipart keys (composite primary keys) (#4734)

* docs: add guide for using multipart keys (composite primary keys)

* remove references

Nauna committed:
Merge pull request #4478 from kgrgreer/parsing-util

PT-FIXES: DOS-793: add locale-aware NumberParser and associated tests for number formatting and parsing

Nauna committed:
Merge pull request #4553 from kgrgreer/groupby-detemintic-model-naming

 PT-FIXES: DOS-1370 generate deterministic model names to prevent conflicts in chained operations

Nauna committed:
Merge pull request #4587 from kgrgreer/set-agent

PT-FIXES: DOS-1432 Add Set sink and enhance existing sinks for multi-row support

abdelaziz-mahdy committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into import-modal-fixes

abdelaziz-mahdy committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into fix-flow-export

abdelaziz-mahdy committed:
Merge with dev

abdelaziz-mahdy committed:
Merge with Dev

abdelaziz-mahdy committed:
Merge with dev

abdelaziz-mahdy committed:
Merge With Dev

abdelaziz-mahdy committed:
feat: add strict validation mode for date parsing with error handling

abdelaziz-mahdy committed:
refactor: streamline date parser tests and enable strict validation mode

abdelaziz-mahdy committed:
feat: implement strict and non-strict validation modes for date parsing

abdelaziz-mahdy committed:
fix: update error message for unsupported date format in strict mode

Nauna committed:
Merge pull request #4694 from kgrgreer/required-fields-can-not-be-empty

PT-FIXES: DOS-1981: implement custom validation for required fields and  track empty source values

Nauna committed:
Merge pull request #4738 from kgrgreer/enable-ndiff-on-flow-dao

Enable NDiff on Flow DAO and Improve Compare UI

Nauna committed:
fix spacing

abdelaziz-mahdy committed:
Revert " PT-FIXES: DOS-1370 generate deterministic model names to prevent conflicts in chained operations"

Nauna committed:
Merge pull request #4594 from kgrgreer/expressions-outputtype

PT-FIXES: DOS-1455 : add outputType property to date expression classes

abdelaziz-mahdy committed:
fix conflicts

abdelaziz-mahdy committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into date-util-date-parser-in-java

abdelaziz-mahdy committed:
feat(date-util): implement lenient and strict validation modes for date parsing

abdelaziz-mahdy committed:
feat(parse): add new date format tests for Java compatibility

- Implemented tests for various date formats to ensure parity with JavaScript date parsing.
- Added tests for timestamps (both 10-digit and 13-digit), single-digit month/day formats, space-separated month names, MM/DD/YY formats, and fractional seconds.

Nauna committed:
Merge pull request #4743 from kgrgreer/revert-4553-groupby-detemintic-model-naming

Revert " PT-FIXES: DOS-1370 generate deterministic model names to prevent conflicts in chained operations"

abdelaziz-mahdy committed:
feat(parse): add Unix Date.toString() format and timestamp support to DateParser

  Add support for parsing Unix/Java Date.toString() format (e.g., "Tue Apr 01 05:17:59 GMT 2025")
  with timezone offset handling. Add Unix timestamps (10-digit seconds, 13-digit milliseconds).
  Make strictValidation static to match JavaScript singleton behavior.

  Remove YYMMDD from main entry point - ambiguous with MMDDYY, use opt_name='yymmdd' explicitly.

Nauna committed:
Merge pull request #4741 from kgrgreer/improve-column-parser

PT-FIXES DOS-2183 : add underscore-to-camelCase normalization in ColumnParser

abdelaziz-mahdy committed:
refactor(reflow): simplify ColumnParser with unified normalization

  Replace multiple literal parser strategies with a single map-based
  lookup. Normalize input (underscores/spaces → camelCase) before lookup,
  reducing code complexity while supporting more input variations.

  Expand test coverage from ~30 to ~75 assertions covering all edge cases.

abdelaziz-mahdy committed:
test(ColumnParser): add tests for non-standard property name normalization

Nauna committed:
Merge pull request #4559 from kgrgreer/date-util-date-parser-in-java

PT-FIXES: DOS-1290 DateParser in java

Nauna committed:
Merge pull request #4618 from kgrgreer/mapping-view

PT-FIXES: DOS-1575 improve Mapping view

Nauna committed:
Merge pull request #4713 from kgrgreer/import-modal-fixes

PT-FIXES: DOS-2000 Import modal redirect and empty pop-up issues

Nauna committed:
Merge pull request #4711 from kgrgreer/fix-flow-export

PT-FIXES: DOS-2004 export functionality and improve error handling

Nauna committed:
Merge pull request #4744 from kgrgreer/improve-column-parser-more

Simplify ColumnParser with dual-side normalization

Nauna committed:
Merge pull request #4739 from kgrgreer/fix-choice-view

fix(choice-view): prevent auto-selection of single choice when placeholder is set

Nauna committed:
Merge pull request #4691 from kgrgreer/add-missing-toProperties-functions

feat: add toProperties functions to Max and Min sinks; improve error handling in LabeledSink

abdelaziz-mahdy committed:
feat(parse): add Julian date format support to DateParser (#4745)

Add YYDDD (5-digit) and YDDD (4-digit) Julian date parsing for
day-of-year formats commonly used in financial data files.

- YYDDD: 2-digit year + 3-digit day (e.g., 25216 = Aug 4, 2025)
- YDDD: 1-digit year with sliding window + 3-digit day
- Combined 'juliandate' format auto-detects 4 or 5 digit input
- Add JULIANDATE option to DateFormat enum for mapping UI
- Implement in both JS and Java with consistent behavior
- YDDD uses sliding window to avoid Y2030 problem

kgrgreer committed:
Add logging when save button is disabled in DAOCreateController.

Nauna committed:
Fix client-side download by delaying Blob URL revocation (#4748)

kgrgreer committed:
Add Auto-complete control.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

abdelaziz-mahdy committed:
fix(view): ensure subscription only occurs if sub method exists (#4749)

abdelaziz-mahdy committed:
feat(reflow): add YYMMDD date format and parserSymbol property to DateFormat enum

Add parserSymbol property to DateFormat enum values to eliminate switch
statements in formatToParserName() methods. New date formats can now be
added by simply defining the enum value with its parser symbol.

Also adds YYMMDD format for 6-digit dates (e.g., 250325 = March 25, 2025).

sarthak-marwaha committed:
Update Some loginView OAuth

Nauna committed:
Merge pull request #4754 from kgrgreer/LoginViewOAuthStyling

LoginView OAuth styling changes

Nauna committed:
Merge pull request #4752 from kgrgreer/add-yymmdd-support

Add YYMMDD date format and parserSymbol property

kgrgreer committed:
Make AutoControl permissionRequired.

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

kgrgreer committed:
Initial check-in of Cells.md.

abdelaziz-mahdy committed:
feat(mapping): enhance header normalization and resolve constant name collisions

abdelaziz-mahdy committed:
fix(reflow): add exact match priority to ColumnParser

Prevents collisions when headers normalize to same value (e.g.,
StartDate vs Start_date both normalizing to startDate). Now tries
exact case-insensitive match first, then falls back to normalized
match with first-match-wins semantics.

Also adds parse() method to SimpleJavaScriptParser for SmartView
compatibility.

abdelaziz-mahdy committed:
fix(mlang): add missing outputType to DateToYYYYMMExpr

kgrgreer committed:
Avoid returning undefined in expression.

kgrgreer committed:
Add ability to convert FBE's to markdown.

Nauna committed:
Merge pull request #4756 from kgrgreer/fix-mapping-conflicts

PT-FIXES: DOS-2182 Fix FOAM constant name collisions in dynamic Mapping

Nauna committed:
Merge pull request #4757 from kgrgreer/improve-coulmn-parser

PT-FIXES: DOS-2182 Add exact match priority to ColumnParser

Nauna committed:
Merge pull request #4758 from kgrgreer/add-missing-output-type

PT-FIXES:DOS-2203 fix(mlang): add missing outputType to DateToYYYYMMExpr

sarthak-marwaha committed:
Fix table row styling when there are no actions in the table row (#4759)

abdelaziz-mahdy committed:
fix(parse): constrain compact date formats to years 1900-2999 (#4755)

Changed mmddyyyycompact, ddmmyyyycompact, and yyyyddmmcompact to use
year4_1900_2999 instead of year4, matching the existing constraint
in yyyymmddcompact.

This prevents false positives when parsing filenames with embedded
dates. For example, "02025033" was incorrectly matching as
MM=02, DD=02, YYYY=5033 instead of being rejected.

Added tests to verify the year constraint works correctly for all
compact date formats.

lchanmann committed:
Fix TranslationConsole view (#4746)

* Fix TranslationConsole view

* Address comment

* Padding with em to avoid gap change on resize

sarthak-marwaha committed:
Ensure flow names are encoded to avoid special chars breaking mementos (#4747)

lchanmann committed:
Oauth enhancement (#4740)

* Show oauth provider icon

* Default to oauth in window

* Store oauth scope as it could vary among different providers

* Enable explicit auto signup configuration

* Allow storing email as username

* Missing comma

* Refactor to provider.signUp to drive auto-signup after successful SSO

* Include oauth provider id in error logs

* Show oauth exception toast notification

- after IdP redirect back to us but the app is unable to link the SSO to existing user, it's better to bring back the LoginView as opposed to staying on service/oauth page with error text.

* Update oauth provider example

* Fix authURL in oauthInWindow mode

* Fix emitting post message for oauthInWindow mode

* Default to oauth in a separate window

* Use cps hash for inline js code

* Restore signInWithOIDC signUp arg

* Add userRegistrationEnabled flag to Theme

* Use imported signInWithOIDC if provided, allow login view wrapper to override the method

* Remove scope from OAuthProvider

* Cleanup

kgrgreer committed:
PT-FIXES: Workaround for DOS-2231 column headers aren't shown when vertically scrolled in GridByView.

kgrgreer committed:
Improvements to AutoPrompt.

Nauna committed:
Merge branch 'development' into allow-flow-mode-to-be-imported

Nauna committed:
update Flow.reflow to be more dynamic and controlled

Nauna committed:
Allow Console to preserve nav state via mount overrides

Nauna committed:
Merge pull request #4735 from kgrgreer/allow-flow-mode-to-be-imported

feat: enhance reflow method to support flow mode logging and dynamic routing

kgrgreer committed:
Add WHERE support to the AutGrammar.

VesnaSUG committed:
DOS-2066 Create Docker-optimized build task

sarthak-marwaha committed:
Fix improper fluent functions in element (#4762)

abdelaziz-mahdy committed:
feat(mlang): add precision support to numeric aggregation sinks (#4761)

* feat(mlang): add precision support to numeric aggregation sinks

Allow controlling decimal places for Sum, Average, Min, and Max sink
output values. Precision property is only visible for numeric properties
(Int, Long, Float, Double). Default value of -1 preserves existing
behavior with no rounding.

* fix(mlang): ensure precision property does not generate Java code

* feat(mlang): add postSet logging for arg1 in AbstractUnarySink, Average, and Sum

* refactor(mlang): remove postSet logging and simplify arg1 property in AbstractUnarySink, Average, and Sum

* refactor(mlang): expose precision config in reflow DAOAgents

Move precision configuration to DAOAgent layer (Min, Max, Avg, Sum)
so users can control decimal places in the reflow UI. Centralize
type checking in AbstractUnarySink.applyPrecision() to handle
non-numeric values consistently across all sinks. Remove redundant
formatValue() methods from Max/Min sinks.

* refactor(mlang): remove generateJava property from precision field in AbstractUnarySink

* fix(bench): update Average constructor call for new precision param

* Update src/foam/mlang/sink/AbstractUnarySink.js

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

---------

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

kgrgreer committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into development

sarthak-marwaha committed:
Fix table row actions not showing (#4764)

abdelaziz-mahdy committed:
fix(mlang): correct precision reference in applyPrecision method

Nauna committed:
Merge pull request #4767 from kgrgreer/fix-precission

fix(mlang): correct precision reference in applyPrecision method

kgrgreer committed:
Add support for agentsParser to AutoGrammar.

sarthak-marwaha committed:
Updated nav hiding logic to correspond to flow mode | Updated flow home button to respect browser back rather than always sending users to flow menu

Nauna committed:
Merge pull request #4772 from kgrgreer/FixNavHidingFlows

Updated nav hiding logic to correspond to flow mode | Updated flow ho…

abdelaziz-mahdy committed:
fix(reflow): correct semaphore usage in UploadSink to prevent hangs (#4774)

sarthak-marwaha committed:
Fix stack view sizing when showNav is false

Nauna committed:
Merge pull request #4775 from kgrgreer/FixNewFlowsSizing

Fix stack view sizing when showNav is false

sarthak-marwaha committed:
Recheck useVariants when switching schemes as theme might have changed (#4776)

abdelaziz-mahdy committed:
fix(reflow): remove hardcoded 2-decimal precision from Cells

The Cell.data adapt function forced all non-integer values to 2 decimal
places, which truncated precision for data that needed more accuracy.
Let values display as-is without forced rounding.

Nauna committed:
Merge pull request #4777 from kgrgreer/remove-hardcoded-percission

PT-FIXES: DOS-2199 Remove hardcoded precision from Cells

sarthak-marwaha committed:
Revert back behaviour until we can find a way to stay in app

Nauna committed:
Merge pull request #4778 from kgrgreer/DOS-2267-home-button-redirects-user-outside-the-application

Revert back behaviour until we can find a way to stay in app

Nauna committed:
Stream files with DigCsvDriver

lchanmann committed:
Remove javax.json library as it has been replaced by jakarta.json

lchanmann committed:
Log error

Nauna committed:
Merge pull request #4779 from kgrgreer/remove-javax.json

Remove javax.json library as it has been replaced by jakarta.json

noodlemoodle committed:
get filters from pred (#4780)

Co-authored-by: ruby <ruby.zhang@payticconnect.com>

Nauna committed:
JSONFObjectFormatter outputProperty to default
